pandas/io/tests/test_parsers.py

```diff
@@ -3461,6 +3462,19 @@ def test_buffer_rd_bytes(self):
             except Exception as e:
                 pass
 
+    def test_read_empty_with_usecols(self):
+        # See gh-12493
+        names = ['Dummy', 'X', 'Dummy_2']
+        usecols = names[1:2] # ['X']
+
+        def check_usecols(stringIO):
+            df = read_csv(stringIO, names=names, usecols=usecols)
+            assert_array_equal(df.columns, usecols)
```

Comment:
NEVER use ``numpy.testing`` imports!!!!! 

ALWAYS use pandas testing functions. These are much more comprehensive and fully test all behavior. It is also confusing to a reader.

 Fully test vs an expected frame and use ``assert_frame_equal``